components/TripForm/index.js

```diff
@@ -0,0 +1,102 @@
+import useSWR from "swr";
+import { useRouter } from "next/router";
+import ConfirmationMessage from "../ConfirmationMessage";
+
+export default function TripForm() {
+  const { mutate } = useSWR("/api/trips");
+
+  const router = useRouter();
+
+  async function handleSubmit(event) {
+    event.preventDefault();
+
+    const formData = new FormData(event.target);
+    const tripData = Object.fromEntries(formData);
+    const response = await fetch("/api/trips", {
+      method: "POST",
+      headers: {
+        "Content-Type": "application/json",
+      },
+      body: JSON.stringify(tripData),
+    });
+    if (!response.ok) {
+      console.error(response.status);
+      return;
+    }
+    mutate();
+    router.push("/confirmation");
+    event.target.reset();
+  }
+
+  return (
+    <>
+      <form onSubmit={handleSubmit}>
+        <fieldset>
+          <label htmlFor="title">Title (max. 30 characters)*</label>
+          <input
+            id="title"
+            name="title"
+            type="text"
+            maxLength="30"
+            required
+          ></input>
+
+          <label htmlFor="startDate">Starting date (dd/mm/yyyy)*</label>
+          <input id="startDate" name="startDate" type="date" required></input>
+
+          <label htmlFor="endDate">Ending date (dd/mm/yyyy)*</label>
+          <input id="endDate" name="endDate" type="date" required></input>
+
+          <label htmlFor="city">City (max. 30 characters)*</label>
```

Comment:
Do you need the htmlFor attribute in react? We don't have but it works totally fine.

Could you explain why you include this attribute? 😊

Reply:
htmlFor links a label and the corresponding input in react. If you click on the label of an input field, you will be directed into the input field.